builder/dockerfile/internals.go

```diff
@@ -328,37 +313,40 @@ func (b *Builder) calcCopyInfo(cmdName, origPath string, allowLocalDecompression
 	}
 	origPath = strings.TrimPrefix(origPath, "."+string(os.PathSeparator))
 
-	context := b.context
+	source := b.source
 	var err error
 	if imageSource != nil {
-		context, err = imageSource.context()
+		source, err = imageSource.context()
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	if context == nil {
+	if source == nil {
 		return nil, errors.Errorf("No context given. Impossible to use %s", cmdName)
 	}
 
 	// Deal with wildcards
 	if allowWildcards && containsWildcards(origPath) {
 		var copyInfos []copyInfo
-		if err := context.Walk("", func(path string, info builder.FileInfo, err error) error {
+		if err := filepath.Walk(source.Root(), func(path string, info os.FileInfo, err error) error {
```

Comment:
Yes, the previous implementation expected it as well. The `Path()` method was just hidden from the interface but the actual copy could never work if it was not implemented.